tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensors.java

```diff
@@ -0,0 +1,186 @@
+package org.tensorflow;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.StringJoiner;
+import org.tensorflow.ndarray.NdArray;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.proto.framework.DataType;
+
+/**
+ * Tensor helper methods.
+ */
+final class Tensors {
+
+  /**
+   * Prevent construction.
+   */
+  private Tensors() {
+  }
+
+  /**
+   * Equivalent to {@link #toString(Tensor, Integer) toString(tensor, null)}.
+   *
+   * @param tensor a tensor
+   * @return the String representation of the tensor
+   */
+  public static String toString(Tensor tensor) {
+    return toString(tensor, null);
+  }
+
+  /**
+   * Returns a String representation of a tensor's data. If the output is wider than {@code
+   * maxWidth} characters, it is truncated and "{@code ...}" is inserted in place of the removed
+   * data.
+   *
+   * @param tensor   a tensor
+   * @param maxWidth the maximum width of the output in characters ({@code null} if unlimited). This
+   *                 limit may surpassed if the first or last element are too long.
+   * @return the String representation of the tensor
+   */
+  public static String toString(Tensor tensor, Integer maxWidth) {
+    if (tensor instanceof RawTensor) {
+      tensor = ((RawTensor) tensor).asTypedTensor();
+    }
+    if (!(tensor instanceof NdArray)) {
+      throw new AssertionError("Expected tensor to extend NdArray.\n" +
+          "actual  : " + tensor + "\n" +
+          "dataType: " + tensor.dataType() + "\n" +
+          "class   : " + tensor.getClass());
+    }
```

Comment:
Since you only accept typed tensors for this method, the endpoint should probably be added to `TType` instead of `Tensor`.

Reply:
@karllessard Is there a way for me to handle all tensors (not just `TType`)? It's okay if not. I'm just asking.

Reply:
@karllessard  How do you create a `RawTensor` that doesn't inherit from `NdArray`? Or more precisely how do you create a plain `Tensor` object that doesn't inherit from `NdArray` without using one of the `org.tensorflow.types` classes?

`RawTensor rawTensor = RawTensor.allocate(TFloat32.class, Shape.of(2, 2), -1);`
Still inherits from `NdArray`.

Reply:
You can't user-side, but all of the `Tensor.fromHandle` methods don't inherit, afaik.  And in places where that's used (Session, somewhere in Operand.asTensor()) if a `asTypedTensor()` is missed somewhere it's nice to not crash when printing.  However, afaik, that's covered by the `tensor = ((RawTensor) tensor).asTypedTensor();` line, so the method should support all Tensors not just TType.